### PR TITLE
feat: create TSTypeQuery node when TSImportType has isTypeOf

### LINKED ISSUE #2998: [RFC] AST structure of TSImportType to be more align with rest of AST
**THIS IS BREAKING CHANGE**

### What is this change about?

`TSImportType` has boolean property `isTypeOf` that is inconsistent with other usages of `typeof` within AST structure.

`isTypeOf` property is unique (not used anywhere else in code)

```ts
type x = typeof import('x')
```
```
TSImportType:
  - isTypeOf: boolean
```

```ts
type x = typeof window
```
```
TSTypeQuery:
  - exprName: Identifier
```

### Why we should change it?

- Improve consistency of ast structure.
- Align generated structure with babel representation
- Allow consumers (eg. prettier) to correctly place comment - [link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAHtgXm3SwgDNsB6AKm0gFt6EZtrLsBLejCAJxgAUAclxCAlCAA0ICBhgdoAZ2SgAhr14QA7gAV1CZSlUAbLarTLpAI16qwAazgwAyhjscoAc2QxeAVzhpOHorOAATMPCAGVUvP1VPOAAxPnpVGHkvZBBVPxgIKRAACxh6YwB1Io54RTcwOGcDao4AN2q0bLBFSxAPRTh+HVtPNOQyE37pACtFXAAhWwcnZ1UmKI84MYnAkBncZw9PYzgART8IeC3jSZA3Xn7ebJI4RTBeDjlCjHfYco4wmBFZAADgADNJvhB+uVbBhst8XgMWptpABHc7wIayQw5RQAWigcHC4UKvDg6I4ZKGCVGSHG1x2-XoHB8-kZh2OZwumzp22kMFUVj+AKBSAATPzbBxjIcAMIQRiqbIvACshT8-QAKoLDPSbi0AgBJKCRWDON4fGAAQRNznQxyu-QAvk6gA)
  ```ts
  typeof /* comment */ import('x')
  ```

### PR DESCRIPTION
**Pull request by armano2**
BREAKING CHANGE:
Changes AST structure in a backwards-incompatible way

fixes #2998

### CODE REVIEW

File: packages/typescript-estree/src/convert.ts [hunk 1]
```diff
@@ -2546,19 +2546,32 @@ export class Converter {
...
+        const range = getRange(node, this.ast);
+        if (node.isTypeOf) {
+          const token = findNextToken(node.getFirstToken()!, node, this.ast)!;
+          range[0] = token.getStart(this.ast);
+        }
+        const result = this.createNode<TSESTree.TSImportType>(node, {
           type: AST_NODE_TYPES.TSImportType,
-          isTypeOf: !!node.isTypeOf,
-          parameter: this.convertChild(node.argument),
+          argument: this.convertChild(node.argument),
```
Comment (armano2) [inline packages/typescript-estree/src/convert.ts:2557]: i renamed this property from parameter to argument as is more aligned with typescript/babel and rest of types

eg, call expression has arguments

File: packages/scope-manager/tests/fixtures/type-declaration/type-query-qualified.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const x = { y: { z: 1 } };
+type T = typeof x.y.z;
+type Unresolved = x.y.z;
```
Comment (armano2) [inline packages/scope-manager/tests/fixtures/type-declaration/type-query-qualified.ts:3]: those test can/should be created added separatelly as they validate execute previously uncovered branches

File: packages/typescript-estree/tests/ast-alignment/utils.ts [hunk 3]
```diff
@@ -151,6 +151,28 @@ export function preprocessBabylonAST(ast: BabelTypes.File): any {
...
+          node.argument = {
+            type: AST_NODE_TYPES.TSLiteralType,
+            literal: node.argument,
+            loc: {
+              start: { ...node.argument.loc.start },
+              end: { ...node.argument.loc.end },
+            },
+            range: [...node.argument.range],
+          };
+        }
```
Comment (armano2) [inline packages/typescript-estree/tests/ast-alignment/utils.ts:174]: i reported this issue to babel https://github.com/babel/babel/issues/12833

File: packages/shared-fixtures/fixtures/typescript/basics/type-import-type.src.ts [hunk 4]
```diff
@@ -1,2 +1,4 @@
 type A = typeof import('A');
-type B = import("B").X<Y>;
+type B = import('B').X<Y>;
+// prettier-ignore
+type C = typeof /* test */ import('A');
```
Comment (armano2) [inline packages/shared-fixtures/fixtures/typescript/basics/type-import-type.src.ts:4]: this PR should allow prettier to handle this case

File: packages/typescript-estree/src/convert.ts (hunk 1 above)
Comment (armano2) [reply packages/typescript-estree/src/convert.ts:2557]: alternativelly we could align this node with import expression and use source

**Review by JoshuaKGoldberg — DISMISSED** [review]
LGTM, thanks!

I'm not high enough confidence in AST dealings to merge myself, so I'll defer to @bradzacher

**Comment by bradzacher** [comment]
yeah this is sound - just needs to wait for the next major

**Review by JoshuaKGoldberg — APPROVED** [review]

**Pull request merged**